homeassistant/components/recorder/migration.py

```diff
@@ -41,20 +41,39 @@ def _apply_update(engine, new_version):
     from sqlalchemy import Table
     from . import models
 
-    if new_version == 1:
-        def create_index(table_name, column_name):
-            """Create an index for the specified table and column."""
-            table = Table(table_name, models.Base.metadata)
-            name = "_".join(("ix", table_name, column_name))
-            # Look up the index object that was created from the models
-            index = next(idx for idx in table.indexes if idx.name == name)
-            _LOGGER.debug("Creating index for table %s column %s",
-                          table_name, column_name)
-            index.create(engine)
-            _LOGGER.debug("Index creation done for table %s column %s",
-                          table_name, column_name)
+    def create_index(table_name, column_name):
```

Comment:
I'd defer to @armills on that one since my PR uses mostly what was already there. I would imagine these functions would only be useful during a migration unless things were getting very complex (say by creating a temporary table and creating an index on that for later use) since, otherwise, the real places index definitions live is in the models?

The HA approach to migrations is pretty new to me so I'm still wrapping my head around it, to be fair.